File: internal/module/metadata/module_management_log.go
// SPDX-FileCopyrightText: 2026-present Brian Wang <wangbuke@gmail.com>
// SPDX-License-Identifier: LGPL-3.0-or-later

package metadata

import (
	"time"

	"github.com/choysum-dev/choysum/pkg/meta"
	"gorm.io/datatypes"
)

type ModuleManagementLog struct {
	meta.BaseModel
	JobId            string         `gorm:"column:job_id;size:64;uniqueIndex"`
	ModuleName       string         `gorm:"column:module_name;size:255;index"`
	Action           string         `gorm:"column:action;size:32;index"`
	OperatorUserId   string         `gorm:"column:operator_user_id;size:64;index"`
	ResultStatus     string         `gorm:"column:result_status;size:32;index"`
	JobCreatedAt     *time.Time     `gorm:"column:job_created_at;index"`
	JobFinishedAt    *time.Time     `gorm:"column:job_finished_at;index"`
	ErrorDomain      string         `gorm:"column:error_domain;size:128"`
	ErrorCode        string         `gorm:"column:error_code;size:128"`
	SummaryJson      datatypes.JSON `gorm:"column:summary_json"`
	LastErrorJson    datatypes.JSON `gorm:"column:last_error_json"`
	ServerInstanceId string         `gorm:"column:server_instance_id;size:128"`
	Attempt          int            `gorm:"column:attempt"`
	MaxAttempts      int            `gorm:"column:max_attempts"`
}

func (ModuleManagementLog) TableName() string {
	return "meta_module_management_log"
}

File: modules/meta/service/models/module_management_log.ts
// SPDX-FileCopyrightText: 2026-present Brian Wang <wangbuke@gmail.com>
// SPDX-License-Identifier: Apache-2.0

import { BaseModel, Field, Model } from '@/core/service';
import { _lt } from '../i18n';

@Model('ModuleManagementLog', {
  application: 'meta',
  tableName: 'meta_module_management_log',
  autoMigrate: false,
})
export default class ModuleManagementLog extends BaseModel {
  @Field({ type: 'varchar', size: 64, unique: true, notNull: true, copy: false, string: _lt('Job', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  JobId: string;

  @Field({ type: 'varchar', size: 255, index: true, notNull: true, string: _lt('Module Name', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  ModuleName: string;

  @Field({ type: 'varchar', size: 32, index: true, notNull: true, string: _lt('Action', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  Action: string;

  @Field({ type: 'varchar', size: 64, index: true, string: _lt('Operator', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  OperatorUserId: string;

  @Field({ type: 'varchar', size: 32, index: true, string: _lt('Result Status', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  ResultStatus: string;

  @Field({ type: 'datetime', index: true, string: _lt('Job Created At', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  JobCreatedAt: Date;

  @Field({ type: 'datetime', index: true, string: _lt('Job Finished At', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  JobFinishedAt: Date;

  @Field({ type: 'varchar', size: 128, string: _lt('Error Domain', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  ErrorDomain: string;

  @Field({ type: 'varchar', size: 128, string: _lt('Error Code', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  ErrorCode: string;

  @Field({ type: 'jsonobject', string: _lt('Summary', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  SummaryJson: Record<string, any>;

  @Field({ type: 'jsonobject', string: _lt('Error Detail', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  LastErrorJson: Record<string, any>;

  @Field({ type: 'varchar', size: 128, string: _lt('Server Instance', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  ServerInstanceId: string;

  @Field({ type: 'int', string: _lt('Attempt', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  Attempt: number;

  @Field({ type: 'int', string: _lt('Max Attempts', { scope: 'meta.model.ModuleManagementLog.fields' }) })
  MaxAttempts: number;
}
